### PR TITLE
Layout + tinting rework

### PR DESCRIPTION
**Pull request by GladOSkar**
In response to #925 

I restructured and cleaned up a lot which turned out not to be necessary for the final solution but might be nice anyways. I don't think i introduced any regressions but that should be tested.

Alternating tint is now anchored to the right by counting the visible blocks beforehand and starting the alternating sequence accordingly on the left.

I barely have any rust experience, please lmk if something could be done better :)

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -199,30 +198,40 @@ macro_rules! map_to_owned (
 
 struct PrintState {
```
Comment (ammgws) [inline src/util.rs:199]: Since has_predecessor no longer exists we can get rid of this struct

Comment (GladOSkar) [reply src/util.rs:199]: I guess we can, i already wondered why that was a struct anyway

**Comment by GladOSkar** [comment]
There ya go

**Comment by ammgws** [comment]
Thanks for adding the comments as well!

**Pull request merged**